ptf/saidebugcounters.py

```diff
@@ -0,0 +1,2194 @@
+        try:
+            stage = SAI_ACL_STAGE_INGRESS
+            bind_points = [SAI_ACL_BIND_POINT_TYPE_PORT]
+            action_types = [SAI_ACL_ACTION_TYPE_PACKET_ACTION]
+            action_drop = SAI_PACKET_ACTION_DROP
+            dip_mask = '255.255.255.0'
+
+            acl_bind_point_type_list = sai_thrift_s32_list_t(
+                count=len(bind_points), int32list=bind_points)
+            acl_action_type_list = sai_thrift_s32_list_t(
+                count=len(action_types), int32list=action_types)
+
+            if dip is not None:
+                dip_ind = True
+                dst_ip = sai_thrift_acl_field_data_t(
+                    data=sai_thrift_acl_field_data_data_t(ip4=dip),
+                    mask=sai_thrift_acl_field_data_mask_t(ip4=dip_mask))
+            else:
+                dip_ind = None
+                dst_ip = None
+            if dmac is not None:
+                dmac_ind = True
+                dst_mac = sai_thrift_acl_field_data_t(
+                    data=sai_thrift_acl_field_data_data_t(mac=dmac),
+                    mask=sai_thrift_acl_field_data_mask_t(mac=mac_mask))
+            else:
+                dmac_ind = None
+                dst_mac = None
+
+            acl_table_id = sai_thrift_create_acl_table(
+                self.client,
+                acl_stage=stage,
+                acl_bind_point_type_list=acl_bind_point_type_list,
+                acl_action_type_list=acl_action_type_list,
+                field_dst_ip=dip_ind,
+                field_dst_mac=dmac_ind)
+            self.assertTrue(acl_table_id != 0)
+
+            action_drop = action
+            packet_action = sai_thrift_acl_action_data_t(
+                parameter=sai_thrift_acl_action_parameter_t(s32=action_drop))
+            acl_entry = sai_thrift_create_acl_entry(
+                self.client,
+                table_id=acl_table_id,
+                action_packet_action=packet_action,
+                field_dst_ip=dst_ip,
+                field_dst_mac=dst_mac)
+            self.assertTrue(acl_entry != 0)
+
+            acl_table_group = sai_thrift_create_acl_table_group(
+                self.client,
+                acl_stage=stage,
+                acl_bind_point_type_list=acl_bind_point_type_list,
+                type=SAI_ACL_TABLE_GROUP_TYPE_PARALLEL)
+            acl_group_member = sai_thrift_create_acl_table_group_member(
+                self.client,
+                acl_table_group_id=acl_table_group,
+                acl_table_id=acl_table_id)
+
+        finally:
+            acl.append({
+                'acl_table': acl_table_id,
+                'acl_entry': acl_entry,
+                'acl_table_group': acl_table_group,
+                'acl_group_member': acl_group_member})
+        return acl
+
+
+class GetDebugCounterEnumValuesCapabilities(BaseDebugCounterClass):
+    ''' SAI query DebugCounter enum values capabilities. '''
+
+    def runTest(self):
+        # Supported SAI IN drop reason list
```

Comment:
Do we have more SAI IN drop reasons (also these that are not supported on our side? If so, should we add the remaining ones?

Reply:
I think all are supported